src/lib/swapper/swappers/ZrxSwapper/endpoints.ts

```diff
@@ -81,28 +75,18 @@ export const zrxApi: Swapper2Api = {
     return adapter.buildSendApiTransaction(buildSendApiTxInput)
   },
 
-  checkTradeStatus: async ({
+  checkTradeStatus: ({
     tradeId,
     txId,
   }): Promise<{ status: TxStatus; buyTxId: string | undefined; message: string | undefined }> => {
-    try {
-      const maybeTradeQuoteMetadata = tradeQuoteMetadata.get(tradeId)
-      if (!maybeTradeQuoteMetadata) {
-        return createDefaultStatusResponse(txId)
-      }
-
-      const adapter = assertGetAdapter(maybeTradeQuoteMetadata.chainId)
-      const tx = await adapter.httpProvider.getTransaction({ txid: txId })
-      const status = getTxStatus(tx)
+    // TODO: it might be smart to pass in the chainId rather than having to pull it out of storage
```

Comment:
If the `txId` we're passing was an *actual* txId in the form `accountId*hash*userAddress*memo` (which it isn't currently), then we could use `selectTxById` to select the txId from the store, then deserialize it and get the ChainId from the AccountId.
Interestingly, we don't have a `deserializeTxIndex` function so would have to write one, effectively the opposite of the existing `serializeTxIndex`.

Note however that in its current form, the `txId` we have here is a tx Hash only, so we would have to update `useTradeExecution` to pass down a serialized Txid.

tl;dr passing the `chainId` down would be a much lower lift to start with, and probably more API-y as well

Reply:
Ah right, in that case it might be wise to rename `txId` to `txHash` to make this clearer. I'll implement that now

Reply:
f1351f5feb06d2e1590a7c21dd736ad3e6460179